2.3.3-RC1 always crashes with "classpath is corrupted"

I tried upgrading to 2.3.3-RC1 and I always get

 ```[error] scalafmt version 2.3.3-RC1 classpath is corrupted```

I've tried running `sbt clean update` and that did not help. This issue is not present with 2.3.2 so I do not believe it is my setup.

Thanks for the great tool, looking forward to using 2.3.3!

My configuration is:

```
version = "2.3.3-RC1"
maxColumn = 120 
rewrite.rules = [PreferCurlyFors, SortModifiers, AvoidInfix, SortImports]
trailingCommas = always
assumeStandardLibraryStripMargin = true
optIn.breakChainOnFirstMethodDot = false
spaces.inImportCurlyBraces = true
includeCurlyBraceInSelectChains = true
spaces.beforeContextBoundColon = Always
// These are copied from  https://scalameta.org/scalafmt/docs/configuration.html#other
rewrite.neverInfix.excludeFilters = [
  until
  to
  by
  eq
  ne
  "should.*"
  "contain.*"
  "must.*"
  in
  ignore
  be
  taggedAs
  thrownBy
  synchronized
  have
  when
  size
  only
  noneOf
  oneElementOf
  noElementsOf
  atLeastOneElementOf
  atMostOneElementOf
  allElementsOf
  inOrderElementsOf
  theSameElementsAs
  // these are new
  throws
  returns
  satisfy
]
```

Oops, sorry I misunderstood the problem. This phenomenon is reproduced even in my environment. If we try to use `2.3.3-RC1` from both `sbt-scalafmt (v2.3.0)` and `scalafmt-cli (v2.3.2)`

<details>
<summary> piece of stacktrace from cli</summary>
```
org.scalafmt.cli.FailedToFormat: /Users/tanishiking/dev/src/github.com/tanishiking/scalafmt/.scalafmt.conf
Caused by: java.lang.RuntimeException: scalafmt version 2.3.3-RC2 classpath is corrupted
        at org.scalafmt.interfaces.ScalafmtReporter.error(ScalafmtReporter.java:38)
        at org.scalafmt.cli.ScalafmtCliReporter.error(ScalafmtCliReporter.scala:15)
        at org.scalafmt.dynamic.ScalafmtDynamic.reportError(ScalafmtDynamic.scala:96)
        at org.scalafmt.dynamic.ScalafmtDynamic.format(ScalafmtDynamic.scala:74)
        at org.scalafmt.cli.ScalafmtDynamicRunner$.handleFile(ScalafmtDynamicRunner.scala:96)
        at org.scalafmt.cli.ScalafmtDynamicRunner$.$anonfun$run$4(ScalafmtDynamicRunner.scala:55)
        at org.scalafmt.cli.ScalafmtDynamicRunner$.$anonfun$run$4$adapted(ScalafmtDynamicRunner.scala:49)
        at scala.collection.immutable.List.foreach(List.scala:392)
        at org.scalafmt.cli.ScalafmtDynamicRunner$.$anonfun$run$3(ScalafmtDynamicRunner.scala:49)
        at scala.util.control.Breaks.breakable(Breaks.scala:42)
        at org.scalafmt.cli.ScalafmtDynamicRunner$.run(ScalafmtDynamicRunner.scala:49)
        at org.scalafmt.cli.Cli$.runWithRunner(Cli.scala:135)
        at org.scalafmt.cli.Cli$.run(Cli.scala:84)
        at org.scalafmt.cli.Cli$.mainWithOptions(Cli.scala:69)
        at org.scalafmt.cli.Cli$.main(Cli.scala:58)
        at org.scalafmt.cli.Cli.main(Cli.scala)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at coursier.bootstrap.launcher.a.a(Unknown Source)
        at coursier.bootstrap.launcher.Launcher.main(Unknown Source)
Caused by: java.lang.NoSuchMethodException: org.scalafmt.Scalafmt.format(java.lang.String, org.scalafmt.config.ScalafmtConfig, scala.collection.immutable.Set)
        at java.lang.Class.getMethod(Class.java:1786)
        at org.scalafmt.dynamic.ScalafmtReflect.<init>(ScalafmtReflect.scala:40)
        at org.scalafmt.dynamic.ScalafmtDynamic.$anonfun$resolveClassPath$1(ScalafmtDynamic.scala:188)
        at scala.util.Try$.apply(Try.scala:213)
        at org.scalafmt.dynamic.ScalafmtDynamic.resolveClassPath(ScalafmtDynamic.scala:186)
        at org.scalafmt.dynamic.ScalafmtDynamic.$anonfun$resolveFormatter$3(ScalafmtDynamic.scala:177)
        at scala.util.Either.flatMap(Either.scala:341)
        at org.scalafmt.dynamic.ScalafmtDynamic.$anonfun$resolveFormatter$1(ScalafmtDynamic.scala:177)
        at scala.util.Try$.apply(Try.scala:213)
        at org.scalafmt.dynamic.utils.ReentrantCache.getOrAddToCache(ReentrantCache.scala:41)
        at org.scalafmt.dynamic.ScalafmtDynamic.resolveFormatter(ScalafmtDynamic.scala:163)
        at org.scalafmt.dynamic.ScalafmtDynamic.$anonfun$resolveConfigWithScalafmt$2(ScalafmtDynamic.scala:142)
        at scala.util.Either.flatMap(Either.scala:341)
        at org.scalafmt.dynamic.ScalafmtDynamic.resolveConfigWithScalafmt(ScalafmtDynamic.scala:139)
        at org.scalafmt.dynamic.ScalafmtDynamic.$anonfun$resolveConfig$4(ScalafmtDynamic.scala:125)
        at scala.util.Try$.apply(Try.scala:213)
        at org.scalafmt.dynamic.utils.ReentrantCache.getOrAddToCache(ReentrantCache.scala:41)
        at org.scalafmt.dynamic.ScalafmtDynamic.resolveConfig(ScalafmtDynamic.scala:124)
        at org.scalafmt.dynamic.ScalafmtDynamic.formatDetailed(ScalafmtDynamic.scala:108)
```
</details>

We have to fix this before cutting a new release (2.3.3), but I'm not sure what causes this...

---

@adampauls 
Sorry, the release process for sbt-scalafmt 2.3.1 seemed to fail... I'm gonna check it this weekend.

- https://repo1.maven.org/maven2/org/scalameta/sbt-scalafmt_2.12_1.0/
- https://travis-ci.org/scalameta/sbt-scalafmt/jobs/642724103

---

@kitbellew 
> what is the process of releasing (apart from tagging a commit)

- You can cut a new release from https://github.com/scalameta/scalafmt/releases/new
- since 2.3.3 will have a lot of changes, it's best to write a detailed release note like this https://github.com/scalameta/scalafmt/releases/tag/v2.3.0-RC2

--- a/scalafmt-benchmarks/src/main/scala-2.12/benchmarks/MacroBenchmark.scala
+++ b/scalafmt-benchmarks/src/main/scala-2.12/benchmarks/MacroBenchmark.scala
@@ -57,7 +57,7 @@ abstract class MacroBenchmark(parallel: Boolean, maxFiles: Int)
   @Benchmark
   def scalafmt(): Unit = {
     files.foreach { file =>
-      Try(Scalafmt.format(file))
+      Try(Scalafmt.formatCode(file))
     }
   }
 

--- a/scalafmt-benchmarks/src/main/scala/org/scalafmt/benchmarks/FormatBenchmark.scala
+++ b/scalafmt-benchmarks/src/main/scala/org/scalafmt/benchmarks/FormatBenchmark.scala
@@ -7,7 +7,7 @@ import org.scalafmt.rewrite.{RedundantBraces, SortImports}
 trait FormatBenchmark {
   def formatRewrite(code: String): String = {
     Scalafmt
-      .format(
+      .formatCode(
         code,
         style = ScalafmtConfig.default.copy(
           rewrite = RewriteSettings(

--- a/scalafmt-benchmarks/src/main/scala/org/scalafmt/benchmarks/MicroBenchmark.scala
+++ b/scalafmt-benchmarks/src/main/scala/org/scalafmt/benchmarks/MicroBenchmark.scala
@@ -46,7 +46,7 @@ abstract class MicroBenchmark(path: String*) extends FormatBenchmark {
 
   @Benchmark
   def scalafmt(): String = {
-    Scalafmt.format(code).get
+    Scalafmt.formatCode(code).get
   }
 
   @Benchmark

--- a/scalafmt-cli/src/main/scala/org/scalafmt/cli/Scalafmt210.scala
+++ b/scalafmt-cli/src/main/scala/org/scalafmt/cli/Scalafmt210.scala
@@ -51,7 +51,7 @@ class Scalafmt210 {
       if (filename.endsWith(".sbt")) SRunner.sbt
       else SRunner.default
     val style = scalafmtStyle.copy(runner = runner)
-    Scalafmt.format(code, style) match {
+    Scalafmt.formatCode(code, style, filename = filename) match {
       case Formatted.Success(formattedCode) => formattedCode
       case error =>
         error match {

--- a/scalafmt-cli/src/main/scala/org/scalafmt/cli/Scalafmt210.scala
+++ b/scalafmt-cli/src/main/scala/org/scalafmt/cli/Scalafmt210.scala
@@ -51,7 +51,7 @@ class Scalafmt210 {
       if (filename.endsWith(".sbt")) SRunner.sbt
       else SRunner.default
     val style = scalafmtStyle.copy(runner = runner)
-    Scalafmt.format(code, style, filename = filename) match {
+    Scalafmt.format(code, style) match {
       case Formatted.Success(formattedCode) => formattedCode
       case error =>
         error match {

--- a/scalafmt-cli/src/main/scala/org/scalafmt/cli/ScalafmtCoreRunner.scala
+++ b/scalafmt-cli/src/main/scala/org/scalafmt/cli/ScalafmtCoreRunner.scala
@@ -80,7 +80,7 @@ object ScalafmtCoreRunner extends ScalafmtRunner {
     val scalafmtConfig =
       if (inputMethod.isSbt || inputMethod.isSc) config.forSbt
       else config
-    val formatResult = Scalafmt.format(
+    val formatResult = Scalafmt.formatCode(
       input,
       scalafmtConfig,
       options.range,

--- a/scalafmt-core/shared/src/main/scala/org/scalafmt/Scalafmt.scala
+++ b/scalafmt-core/shared/src/main/scala/org/scalafmt/Scalafmt.scala
@@ -24,6 +24,7 @@ object Scalafmt {
   private val WindowsLineEnding = "\r\n"
   private val UnixLineEnding = "\n"
 
+  // XXX: don't modify signature, scalafmt-dynamic expects it via reflection
   /**
     * Format Scala code using scalafmt.
     *
@@ -43,6 +44,15 @@ object Scalafmt {
       style: ScalafmtConfig,
       range: Set[Range],
       filename: String
+  ): Formatted = {
+    formatCode(code, style, range, filename)
+  }
+
+  private[scalafmt] def formatCode(
+      code: String,
+      style: ScalafmtConfig = ScalafmtConfig.default,
+      range: Set[Range] = Set.empty,
+      filename: String = "<input>"
   ): Formatted = {
     try {
       val runner = style.runner
@@ -82,12 +92,13 @@ object Scalafmt {
     }
   }
 
+  // XXX: don't modify signature, scalafmt-dynamic expects it via reflection
   def format(
       code: String,
       style: ScalafmtConfig = ScalafmtConfig.default,
       range: Set[Range] = Set.empty[Range]
   ): Formatted = {
-    format(code, style, range, "<input>")
+    formatCode(code, style, range)
   }
 
   def parseHoconConfig(configString: String): Configured[ScalafmtConfig] =

--- a/scalafmt-core/shared/src/main/scala/org/scalafmt/Scalafmt.scala
+++ b/scalafmt-core/shared/src/main/scala/org/scalafmt/Scalafmt.scala
@@ -40,9 +40,9 @@ object Scalafmt {
     */
   def format(
       code: String,
-      style: ScalafmtConfig = ScalafmtConfig.default,
-      range: Set[Range] = Set.empty,
-      filename: String = "<input>"
+      style: ScalafmtConfig,
+      range: Set[Range],
+      filename: String
   ): Formatted = {
     try {
       val runner = style.runner
@@ -80,6 +80,14 @@ object Scalafmt {
       // TODO(olafur) add more fine grained errors.
       case NonFatal(e) => Formatted.Failure(e)
     }
+  }
+
+  def format(
+      code: String,
+      style: ScalafmtConfig = ScalafmtConfig.default,
+      range: Set[Range] = Set.empty[Range]
+  ): Formatted = {
+    format(code, style, range, "<input>")
   }
 
   def parseHoconConfig(configString: String): Configured[ScalafmtConfig] =

--- a/scalafmt-core/shared/src/test/scala/org/scalafmt/ScalafmtTest.scala
+++ b/scalafmt-core/shared/src/test/scala/org/scalafmt/ScalafmtTest.scala
@@ -10,7 +10,7 @@ class ScalafmtTest extends scalatest.funsuite.AnyFunSuite {
       config: ScalafmtConfig = ScalafmtConfig.default
   ): Unit = {
     test(logger.revealWhitespace(original).take(30)) {
-      val obtained = Scalafmt.format(original, config).get
+      val obtained = Scalafmt.formatCode(original, config).get
       if (obtained != expected) logger.elem(obtained)
       assert(obtained == expected)
     }

--- a/scalafmt-tests/src/test/scala/org/scalafmt/FidelityTest.scala
+++ b/scalafmt-tests/src/test/scala/org/scalafmt/FidelityTest.scala
@@ -43,7 +43,7 @@ class FidelityTest extends AnyFunSuite with FormatAssertions {
   examples.foreach { example =>
     test(example.filename) {
       val formatted =
-        Scalafmt.format(example.code, filename = example.filename).get
+        Scalafmt.format(example.code, ScalafmtConfig.default).get
       assertFormatPreservesAst(example.code, formatted)(
         scala.meta.parsers.Parse.parseSource,
         Scala211

--- a/scalafmt-tests/src/test/scala/org/scalafmt/FidelityTest.scala
+++ b/scalafmt-tests/src/test/scala/org/scalafmt/FidelityTest.scala
@@ -43,7 +43,7 @@ class FidelityTest extends AnyFunSuite with FormatAssertions {
   examples.foreach { example =>
     test(example.filename) {
       val formatted =
-        Scalafmt.format(example.code, ScalafmtConfig.default).get
+        Scalafmt.formatCode(example.code, filename = example.filename).get
       assertFormatPreservesAst(example.code, formatted)(
         scala.meta.parsers.Parse.parseSource,
         Scala211

--- a/scalafmt-tests/src/test/scala/org/scalafmt/FormatTests.scala
+++ b/scalafmt-tests/src/test/scala/org/scalafmt/FormatTests.scala
@@ -52,22 +52,19 @@ class FormatTests
 
   def run(t: DiffTest, parse: Parse[_ <: Tree]): Unit = {
     val runner = scalafmtRunner(t.style.runner).copy(parser = parse)
-    val obtained = Scalafmt.format(
-      t.original,
-      t.style.copy(runner = runner),
-      filename = t.filename
-    ) match {
-      case Formatted.Failure(e)
-          if t.style.onTestFailure.nonEmpty && e.getMessage.contains(
-            e.getMessage
-          ) =>
-        t.expected
-      case Formatted.Failure(e: Incomplete) => e.formattedCode
-      case Formatted.Failure(e: SearchStateExploded) =>
-        logger.elem(e)
-        e.partialOutput
-      case x => x.get
-    }
+    val obtained =
+      Scalafmt.format(t.original, t.style.copy(runner = runner)) match {
+        case Formatted.Failure(e)
+            if t.style.onTestFailure.nonEmpty && e.getMessage.contains(
+              e.getMessage
+            ) =>
+          t.expected
+        case Formatted.Failure(e: Incomplete) => e.formattedCode
+        case Formatted.Failure(e: SearchStateExploded) =>
+          logger.elem(e)
+          e.partialOutput
+        case x => x.get
+      }
     debugResults += saveResult(t, obtained, onlyOne)
     if (t.style.rewrite.rules.isEmpty &&
       !t.style.assumeStandardLibraryStripMargin &&
@@ -77,13 +74,8 @@ class FormatTests
         t.style.runner.dialect
       )
     }
-    val formattedAgain = Scalafmt
-      .format(
-        obtained,
-        t.style.copy(runner = runner),
-        filename = t.filename
-      )
-      .get
+    val formattedAgain =
+      Scalafmt.format(obtained, t.style.copy(runner = runner)).get
 //          getFormatOutput(t.style, true) // uncomment to debug
     assertNoDiff(formattedAgain, obtained, "Idempotency violated")
     if (!onlyManual) {

--- a/scalafmt-tests/src/test/scala/org/scalafmt/FormatTests.scala
+++ b/scalafmt-tests/src/test/scala/org/scalafmt/FormatTests.scala
@@ -52,19 +52,22 @@ class FormatTests
 
   def run(t: DiffTest, parse: Parse[_ <: Tree]): Unit = {
     val runner = scalafmtRunner(t.style.runner).copy(parser = parse)
-    val obtained =
-      Scalafmt.format(t.original, t.style.copy(runner = runner)) match {
-        case Formatted.Failure(e)
-            if t.style.onTestFailure.nonEmpty && e.getMessage.contains(
-              e.getMessage
-            ) =>
-          t.expected
-        case Formatted.Failure(e: Incomplete) => e.formattedCode
-        case Formatted.Failure(e: SearchStateExploded) =>
-          logger.elem(e)
-          e.partialOutput
-        case x => x.get
-      }
+    val obtained = Scalafmt.formatCode(
+      t.original,
+      t.style.copy(runner = runner),
+      filename = t.filename
+    ) match {
+      case Formatted.Failure(e)
+          if t.style.onTestFailure.nonEmpty && e.getMessage.contains(
+            e.getMessage
+          ) =>
+        t.expected
+      case Formatted.Failure(e: Incomplete) => e.formattedCode
+      case Formatted.Failure(e: SearchStateExploded) =>
+        logger.elem(e)
+        e.partialOutput
+      case x => x.get
+    }
     debugResults += saveResult(t, obtained, onlyOne)
     if (t.style.rewrite.rules.isEmpty &&
       !t.style.assumeStandardLibraryStripMargin &&
@@ -74,8 +77,13 @@ class FormatTests
         t.style.runner.dialect
       )
     }
-    val formattedAgain =
-      Scalafmt.format(obtained, t.style.copy(runner = runner)).get
+    val formattedAgain = Scalafmt
+      .formatCode(
+        obtained,
+        t.style.copy(runner = runner),
+        filename = t.filename
+      )
+      .get
 //          getFormatOutput(t.style, true) // uncomment to debug
     assertNoDiff(formattedAgain, obtained, "Idempotency violated")
     if (!onlyManual) {

--- a/scalafmt-tests/src/test/scala/org/scalafmt/util/HasTests.scala
+++ b/scalafmt-tests/src/test/scala/org/scalafmt/util/HasTests.scala
@@ -192,13 +192,8 @@ trait HasTests extends AnyFunSuiteLike with FormatAssertions {
 
   def defaultRun(t: DiffTest, parse: Parse[_ <: Tree]): Unit = {
     val runner = scalafmtRunner(t.style.runner).copy(parser = parse)
-    val obtained = Scalafmt
-      .format(
-        t.original,
-        t.style.copy(runner = runner),
-        filename = t.filename
-      )
-      .get
+    val obtained =
+      Scalafmt.format(t.original, t.style.copy(runner = runner)).get
     if (t.style.rewrite.rules.isEmpty) {
       assertFormatPreservesAst(t.original, obtained)(
         parse,

--- a/scalafmt-tests/src/test/scala/org/scalafmt/util/HasTests.scala
+++ b/scalafmt-tests/src/test/scala/org/scalafmt/util/HasTests.scala
@@ -192,8 +192,13 @@ trait HasTests extends AnyFunSuiteLike with FormatAssertions {
 
   def defaultRun(t: DiffTest, parse: Parse[_ <: Tree]): Unit = {
     val runner = scalafmtRunner(t.style.runner).copy(parser = parse)
-    val obtained =
-      Scalafmt.format(t.original, t.style.copy(runner = runner)).get
+    val obtained = Scalafmt
+      .formatCode(
+        t.original,
+        t.style.copy(runner = runner),
+        filename = t.filename
+      )
+      .get
     if (t.style.rewrite.rules.isEmpty) {
       assertFormatPreservesAst(t.original, obtained)(
         parse,

